### PR TITLE
Allow search params when converting bsky.app urls

### PR DESCRIPTION
**Pull request by mozzius**
Allows linking to specific searches, i.e. https://bsky.app/profile/jgarplind.bsky.social/post/3kuyshu7heu2a

### CODE REVIEW

File: src/lib/strings/url-helpers.ts [hunk 1]
```diff
@@ -162,7 +162,7 @@ export function convertBskyAppUrlIfNeeded(url: string): string {
   if (isBskyAppUrl(url)) {
     try {
       const urlp = new URL(url)
-      return urlp.pathname
+      return urlp.pathname + urlp.search
```
Comment (gaearon) [inline src/lib/strings/url-helpers.ts:165]: I'm not sure this is a safe change — some code calling `convertBskyAppUrlIfNeeded` assumes there's no search. What do you think of the callsites? Could make a separate function for the case of handling in-app links. Or could return a richer data structure and let the caller decide which parts to use.

**Comment by gaearon** [comment]
Doesn't seem to fix it for native btw

**Review by gaearon — CHANGES_REQUESTED** [review]
Looks like this would fix it for web but it still doesn't work on native. I think for native we might want to have a pushable version of the Search. This might be useful in other places too, like for searches themselves. Let's maybe leave this for now.

**Comment by mozzius** [comment]
I wish we were using Expo Router, this would *just work*

**Comment by haileyok** [comment]
Don't we just need to add `Search` into `commonScreens` instead of only being inside the `Search` tab? @mozzius

**Comment by pfrazee** [comment]
I'll take a stab at it

**Pull request closed without merging**